File: HandZ6Api/src/main/java/org/example/handz6api/model/Usuario.java
package org.example.handz6api.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "usuario")
public class Usuario {
    @Id
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "username", nullable = false, length = 50)
    private String username;

    @Column(name = "password", nullable = false)
    private String password;

    @Column(name = "email", length = 100)
    private String email;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "rol_id", nullable = false)
    private Rol rol;

    @ManyToOne(fetch = FetchType.LAZY)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "jugador_id")
    private Jugador jugador;
    @ManyToOne(fetch = FetchType.LAZY)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "equipo_id")
    private Equipo equipo;


}

File: HandZ6/app/src/main/java/com/example/handz6/data/Entities/Usuario.kt
package com.example.handz6.data.Entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey


// Usuario.kt
@Entity(
    tableName = "usuario",
    foreignKeys = [
        ForeignKey(
            entity = Rol::class,
            parentColumns = ["id"],
            childColumns = ["rol_id"],
            onDelete = ForeignKey.NO_ACTION
        ),
        ForeignKey(
            entity = Jugador::class,
            parentColumns = ["id"],
            childColumns = ["jugador_id"],
            onDelete = ForeignKey.SET_NULL
        ),
        ForeignKey(
            entity = Equipo::class,
            parentColumns = ["id"],
            childColumns = ["equipo_id"],
            onDelete = ForeignKey.SET_NULL  // 👈 añade esto
        )
    ],
    indices = [
        Index(value = ["rol_id"]),
        Index(value = ["jugador_id"]),
        Index(value = ["equipo_id"])  // 👈 añade esto
    ]
)
data class Usuario(
    @PrimaryKey
    @ColumnInfo(name = "id")
    val id: Long,

    @ColumnInfo(name = "username")
    val username: String,

    @ColumnInfo(name = "password")
    val password: String,

    @ColumnInfo(name = "email")
    val email: String? = null,

    @ColumnInfo(name = "rol_id")
    val rolId: Long,

    @ColumnInfo(name = "jugador_id")
    val jugadorId: Long? = null,

    @ColumnInfo(name = "equipo_id")
    val equipoId: Long? = null  // null si es administrador
)
